tec-list-parse.py

```diff
@@ -8,103 +8,172 @@
+        doc = minidom.parse(Url)
+        currencies = doc.getElementsByTagName('valuta')
+        currency_name = [elem.getElementsByTagName('opis')[0].firstChild.data
+                         for elem in currencies]
+        currency_value = [elem.getElementsByTagName(erste_rate_type)[0].firstChild.data
+                          for elem in currencies]
+
+    currency_list = OrderedDict(zip(currency_name, currency_value))
+
+    for name, value in currency_list.items():
+        if currency == 'all':
+            print('{}\t{}'.format(name, value))
+        elif currency.lower() == name.lower():
+            print('{}\t{}'.format(name, value))
 
 
 def doit():
-    """
+    '''
     Set up the available program options
     Call the proper functions with proper parameters depending on user
     input
-    """
+    '''
     parser = argparse.ArgumentParser()
     parser.add_argument('-a', '--all', help='Show values for all available \
-            currencies', action="store_true", dest='All')
+            currencies', action='store_true', dest='all')
     parser.add_argument('-c', '--currency', help='Set the currency for which \
             the value is shown', type=str, dest='currency')
+    parser.add_argument('-t', '--type', help='Set the currency rate type for \
+            which the value is shown', type=str, dest='rate_type', default='mean')
     parser.add_argument('-s' '--source', help='Set the source from which the \
             data is retrieved', type=str, dest='source')
 
     args = parser.parse_args()
 
-    if not args.All and not args.currency and not args.source:
+    if not args.all and not args.currency and not args.source:
         parser.print_help()
 
-    if args.All and not args.source:
-        getPBZData('all')
-        getHNBData('all')
-    elif args.All and args.source:
-        if args.source == 'HNB':
-            getHNBData('all')
-        if args.source == 'PBZ':
-            getPBZData('all')
+    if args.all and not args.source:
+        get_pbz_data('all', args.rate_type)
+        get_erste_data('all', args.rate_type)
+        get_hnb_data('all', args.rate_type)
+    elif args.all and args.source:
+        if args.source.lower() == 'HNB'.lower():
+            get_hnb_data('all', args.rate_type)
+        if args.source.lower() == 'PBZ'.lower():
+            get_pbz_data('all', args.rate_type)
+        if args.source.lower() == 'ERSTE'.lower():
+            get_erste_data('all', args.rate_type)
     elif args.currency and not args.source:
-        getPBZData(args.currency)
-        getHNBData(args.currency)
+        get_pbz_data(args.currency, args.rate_type)
+        get_erste_data(args.currency, args.rate_type)
+        get_hnb_data(args.currency, args.rate_type)
     elif args.currency and args.source:
-        if args.source == 'HNB':
-            getHNBData(args.currency)
-        if args.source == 'PBZ':
-            getPBZData(args.currency)
+        if args.source.lower() == 'HNB'.lower():
```

Comment:
Why not just compare with 'hnb'? Is there really need to call an additional method?